src/megatron/bridge/models/conversion/model_bridge.py

```diff
@@ -125,11 +126,12 @@ def _megatron_local_name_to_global(
         _, layer_module = get_module_and_param_from_name(models=models, param_name=layer_prefix, vp_stage=vp_stage)
 
         local_layer_number = int(param_name.split("layers.")[1].split(".")[0])
-        global_layer_number = layer_module.layer_number - 1
-        param_name = param_name.replace(
-            f"layers.{local_layer_number}.",
-            f"layers.{global_layer_number}.",
-        )
+        if isinstance(layer_module, TransformerLayer):
```

Comment:
why is this needed?

Reply:
Non Mcore models does not neccessary have this attribute: e.g. The vision encoder for gemma3 (even tho it is not sharded)